sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/jms/AbstractServiceBusJMSAutoConfiguration.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.jms;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.jms.DefaultJmsListenerContainerFactoryConfigurer;
+import org.springframework.boot.autoconfigure.jms.JmsProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jms.config.DefaultJmsListenerContainerFactory;
+import org.springframework.jms.config.JmsListenerContainerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.jms.ConnectionFactory;
+
+/**
+ * Abstract autoconfiguration class of ServiceBusJMS for JmsListenerContainerFactory.
+ */
+@Configuration
+public abstract class AbstractServiceBusJMSAutoConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractServiceBusJMSAutoConfiguration.class);
+
+    final JmsProperties jmsProperties;
+    final AzureServiceBusJMSProperties azureServiceBusJMSProperties;
+
+    public AbstractServiceBusJMSAutoConfiguration(JmsProperties jmsProperties,
+                                                  AzureServiceBusJMSProperties azureServiceBusJMSProperties) {
+        this.jmsProperties = jmsProperties;
+        this.azureServiceBusJMSProperties = azureServiceBusJMSProperties;
+    }
+
+    @PostConstruct
+    private void validate() {
+        AzureServiceBusJMSProperties.Listener listener = azureServiceBusJMSProperties.getListener();
+        if ((listener.isSubscriptionDurable() || listener.isSubscriptionShared()) && !jmsProperties.isPubSubDomain()) {
```

Comment:
I have a question:
if user configured jmsProperties.isPubSubDomain=false, but doesn't set listener's subscriptionDurable, because subscriptionDurable is true by default, is it expected?

Reply:
No, I will reconsider the default configuration here, thx.